STM32Cube/Tasks/vn_handler.c

```diff
@@ -85,10 +84,9 @@ static void send3VectorStateCanMsg_double(uint64_t time, double vector[3], uint8
 
 void vnIMUHandler(void *argument)
 {
-	//HAL_UART_RegisterCallback(&huart4, HAL_UART_RX_COMPLETE_CB_ID, USART1_DMA_Rx_Complete_Callback); //TODO: use user-defined callback binding
+	HAL_UART_RegisterRxEventCallback(&huart1, VN_UASART1_RX_callback);
```

Comment:
this should be done in the init method so we can check it that it succeeds like the other init actions

Reply:
I've added an init method and moved it there? All the other init methods return void so I'm not sure what you mean exactly?

Reply:
Do they? i've been modifying all the inits to return bool so they can be checked at startup
some are still void though (WIP)